tools/generator/src/Logger.swift

```diff
@@ -33,22 +33,24 @@ final class DefaultLogger<E: TextOutputStream, O: TextOutputStream>: Logger {
     }
 
     func logDebug(_ message: @autoclosure () -> String) {
-        // TODO: Colorize
```

Comment:
Thanks for doing this!

One of the main reasons I didn't tackle this yet, was I wanted to make sure we _didn't_ colorize in certain cases. For a long time I believed that would correlate with Bazel's setting, but then I realized that the output would be cached so maybe it shouldn't? Then I realized we don't cache the generation for various reasons.

All of that is to say I still think we should conditionally do this. I think we should have two implementations, versus a bool. And for the condition, ideally we can detect the color mode for Bazel, but if not, we can add a Starlark build setting ourselves. The main reason I think we should allow it to be turned off, is some people run this on CI that doesn't support color codes, and it will make the output all funny for them.

Reply:
I started writing a utility that would detect if the generator was running in a TTY that supported colors, but ultimately that wouldn't work because Bazel doesn't pass through the `TERM` environment variable.

So after speaking with @brentleyjones , he suggested that we add an `xcodeproj` macro argument for `colorize`, which is now what I've done.